metaflow/cli.py

```diff
@@ -338,8 +338,8 @@ def logs(obj, input_path, stdout=None, stderr=None, both=None, timestamps=False)
     )
     if task_id:
         ds_list = [
-            obj.datastore(
-                obj.flow.name,
+            TaskDataStore(
+                obj.flow_datastore,
```

Comment:
@zhugejun The signature for the constructor is different for TaskDataStore. This PR actually fails to address #816 at the moment. 

```
Traceback (most recent call last):
  File "/Users/savin/Code/metaflow/metaflow/cli.py", line 1110, in main
    start(auto_envvar_prefix="METAFLOW", obj=state)
  File "/Users/savin/opt/miniconda3/envs/py3/lib/python3.7/site-packages/click/core.py", line 1128, in __call__
    return self.main(args, kwargs)
  File "/Users/savin/opt/miniconda3/envs/py3/lib/python3.7/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/Users/savin/opt/miniconda3/envs/py3/lib/python3.7/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/savin/opt/miniconda3/envs/py3/lib/python3.7/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, ctx.params)
  File "/Users/savin/opt/miniconda3/envs/py3/lib/python3.7/site-packages/click/core.py", line 754, in invoke
    return __callback(args, kwargs)
  File "/Users/savin/opt/miniconda3/envs/py3/lib/python3.7/site-packages/click/decorators.py", line 38, in new_func
    return f(get_current_context().obj, args, kwargs)
  File "/Users/savin/Code/metaflow/metaflow/cli.py", line 347, in logs
    allow_unsuccessful=True,
TypeError: __init__() got an unexpected keyword argument 'allow_unsuccessful'
```